activiti-api-impl/activiti-api-task-runtime-impl/src/main/java/org/activiti/runtime/api/impl/TaskRuntimeHelper.java

```diff
@@ -184,5 +192,59 @@ private String getAuthenticatedUser() {
         }
         return internalTask;
     }
+    
+    public Map<String, org.activiti.engine.impl.persistence.entity.VariableInstance> getInternalTaskVariables(String taskId) {
+    	return taskService.getVariableInstancesLocal(taskId);
+    }
+    
+    public void createVariable(boolean isAdmin, CreateTaskVariablePayload createTaskVariablePayload) {	
+    	if (!isAdmin) {
+    		 //Check assignee
```

Comment:
@CTI777 @erdemedeiros I think that all user methods should check the assignee. If you are not the assignee of the task you shouldn't be able to see, create or modify variables. So yes.. we should check for assignee here.